src/haz3lcore/dynamics/Evaluator.re

```diff
@@ -1271,7 +1327,49 @@ and evaluate_test_eq =
   let* arg_result = evaluate(env, arg_show);
 
   (arg_show, arg_result) |> return;
-};
+}
+and module_evaluate:
+  (ClosureEnvironment.t, Environment.t, DHExp.t) => m(EvaluatorResult.t) =
+  (env, env_in, d) => {
+    switch (d) {
+    | Let(dp, d1, d2) =>
+      let* r1 = evaluate(env, d1);
+      switch (r1) {
+      | BoxedValue(d1)
+      | Indet(d1) =>
+        switch (matches(dp, d1)) {
+        | IndetMatch
+        | DoesNotMatch => Indet(Closure(env, Let(dp, d1, d2))) |> return
+        | Matches(env') =>
+          let env = evaluate_extend_env(env', env);
+          let env_in =
+            Environment.union(env', env_in)
+            |> ClosureEnvironment.of_environment;
```

Comment:
I think the   `|> ClosureEnvironment.of_enviornment |> ClosureEnvironment.map_of` is redundant here

Reply:
Resolved. Thanks!